dask/dataframe/core.py

```diff
@@ -336,6 +335,25 @@ def quantiles(self, q):
         """
         return quantiles(self, q)
 
+    def resample(self, rule, how=None, axis=0, fill_method=None, closed=None,
+                 label=None, convention='start', kind=None, loffset=None,
+                 limit=None, base=0):
+        first, last = self.divisions[0], self.divisions[-1]
+        new_divisions = list(pd.date_range(start=first, end=last, freq=rule))
```

Comment:
Another issue here is that `divisions` is left inclusive.  A division corresponds to the start / minimum value of a partition.  

`pd.date_range` seems to give ending values for frequencies like months, at least by default.

```python
In [4]: pd.date_range('2014-01-01', '2015-05-23', freq='M')
Out[4]: 
DatetimeIndex(['2014-01-31', '2014-02-28', '2014-03-31', '2014-04-30',
               '2014-05-31', '2014-06-30', '2014-07-31', '2014-08-31',
               '2014-09-30', '2014-10-31', '2014-11-30', '2014-12-31',
               '2015-01-31', '2015-02-28', '2015-03-31', '2015-04-30'],
              dtype='datetime64[ns]', freq='M', tz=None)
```